### PR TITLE
[Merged by Bors] - Initial stuff for producer smartengine feature flag

### LINKED ISSUE #1788: Add Optional smartstream producer
Similar to how the [consume has an optional `wasm_module`](https://github.com/infinyon/fluvio/blob/0e7d8c7c9fbb180e13828789dbb3cabf01114050/crates/fluvio/src/consumer.rs#L683), which is sent to the SPU on the backend, we should add the ability to produce with a smartmodule before a/an record(s) sent to the SPU.

This will need to be a feature flag addition as it `fluvio-smartengine` will not compile to the wasm-backend (I think). This should be a feature flag which is off by default and isn't available in the wasm32 build.

### PR DESCRIPTION
**Pull request by simlay**
Closes #1788.

### CODE REVIEW

File: crates/fluvio/src/producer/mod.rs [hunk 1]
```diff
@@ -88,10 +165,32 @@ impl TopicProducer {
...
+                use dataplane::smartstream::SmartStreamInput;
+                use std::convert::TryFrom;
+                if let Some(smart_payload) = &self.smartstream_config.wasm_module {
+
+
+                    let engine = SmartEngine::default();
+                    let mut smartstream = engine.create_module_from_payload(smart_payload.clone()).map_err(|e| FluvioError::Other(format!("SmartEngine - {:?}", e)))?;
+
+                    let output = smartstream.process(SmartStreamInput::try_from(entries)?).map_err(|e| FluvioError::Other(format!("SmartEngine - {:?}", e)))?;
+                    entries = output.successes;
```
[the remark this answers is not in the document]
Comment (morenol) [reply crates/fluvio/src/producer/mod.rs:185]: Yes, I see this in the SPU: 

Batch is too big: BatchTooBig(
    1048588,
)

**Review by morenol — APPROVED** [review]
lgtm, the only problem that I see is that when using array_map we could have problems with batch size

**Review by sehz — APPROVED** [review]
LGTM

**Pull request closed without merging**